Add ability to specify multiple lambda handlers per project.

dickwall opened the pull request:
By specifying a sequence of pairs from lambda function name to handler
method, in the `lambdaHandlers` setting, you can create or update multiple
lambda functions with a single create or update operation. When used,
`lambdaHandlers` overrides the single lambda handler specified through
the `lambdaName` and `handlerName` settings, but if `lambdaHandlers` is
ignored, the plugin behaves exactly as before, hence retaining backwards
compatibility with existing plugin usages.

fiadliel commented:
It's mostly compatible, but I think we should bump minor version anyway because of the change of return type for the Tasks.

What do you think of changing the type for `createLambda` and `updateLambda` from `taskKey[Seq[LambdaARN]]` to `taskKey[Map[String,LambdaARN]]`, so that we can match up the created Lambdas exactly with the inputs?

dickwall commented:
Changed, and also I cleaned up the lambda-handler code a bit more and unified it further. It never quite sat right just trying to keep the names and handlers ordered while putting them through different functions, so now both the create and update use the same resolver, which returns a map of LambdaName to HandlerName now, and simplified some of the other code as a result.

fiadliel commented:
I'll have a look tomorrow; but to note, we had whisk(e)y tasting tonight, so... 

P.S. Scalawags is great! Also somebody beside me reminisces about Java Posse...

dickwall commented:
Hi @fiadliel, we are just wondering if there are any more requirements for this PR - the changes would be useful on the project we are working on, and it would be great if people could get them from the central repo when they check out.

Thanks (and thanks for the kind words about the Posse - good old days :-) )

stjohnb commented:
:+1: Tying the lambda names and handlers together is a good improvement.

I have opened a 2nd pull request which builds on top of this one with two minor improvements: https://github.com/gilt/sbt-aws-lambda/pull/10

fiadliel commented:
LG, merging. Sorry for the delay.

fiadliel commented:
Oh, and thanks a lot for the PR!
